R/R/consortia_internal_tracking.R

```diff
@@ -0,0 +1,219 @@
+# Synapse data tracking - track "working" data available to internal consortia. 
+# Note modified, new and deleted files.
+# Author: Kelsey Montgomery
+# Date: 2019.08.01
+#'
+#' Get Synapse view - fileview or table
+#'
+#' @param fileview_id A synId c().
+#' @export
+#' @return A tibble.
+#' @examples
+#' \dontrun{
+#' fv <- get_view("syn18691012")
+#' table <- get_view("syn20555115")
+#' }
+#' 
+get_view <- function(fileview_id) {
+  fileview <- synapser::synTableQuery(paste0("SELECT * FROM ", fileview_id))
+  readr::read_csv(fileview$filepath, col_types = readr::cols(.default = "c"))
+}
+#' Check whether updates need to be made
+#'
+#' @param fv A tibble. A fileview imported with [get_view()].
+#' @param table A tibble. A table imported with [get_view()].
+#' @param fileview_id A synId c().
+#' @export
+#' @return If no changes are to be made the string "No changes!" is returned. If 
+#' there are changes to be made to the existing table, a Synapse Table object is 
+#' returned with metadata to describe the upload.
+#' @examples
+#' \dontrun{
+#' update_table(fv = get_view("synId"), 
+#' table = get_view("synId"), 
+#' fileview_id = "syn20555115")
+#' }
+#'
+update_table <- function(fv, table, fileview_id) {
+  if (check_etag(fv, table) == c("No changes to existing annotations")
+      && check_version(fv, table) == c("No changes to existing files")
+      && setequal(fv$id, table$id[!(table$notes %in% c("File removed"))])) {
+    return("No changes!")
+  } else {
+    mod_table(fv, table, fileview_id)
+  }
+}
+#' Bind new files to the existing table, notes deleted files and file/annotation 
```

Comment:
This is a good description for the function. Tidyverse style recommends a separate [title and description](https://style.tidyverse.org/documentation.html#title-and-description), otherwise the same (long) text gets used for both:
<img width="355" alt="Screen Shot 2019-09-10 at 12 33 23 PM" src="https://user-images.githubusercontent.com/4452678/64644351-5a648300-d3c7-11e9-8494-127f61bc8a46.png">